packages/frontend/src/components/QrReader/index.tsx

```diff
@@ -34,352 +41,482 @@ type Props = {
  */
 const SCAN_QR_INTERVAL_MS = 1000 / 30
 
-const worker = new QrWorker() as Worker
-
-export default function QrReader({ onError, onScanSuccess }: Props) {
-  const tx = useTranslationFunction()
-  const { openContextMenu } = useContext(ContextMenuContext)
+export type QrCodeScanRef = {
+  handlePasteFromClipboard: () => void
+}
 
-  const videoRef = useRef<HTMLVideoElement>(null)
-  const canvasRef = useRef<OffscreenCanvas>(new OffscreenCanvas(640, 480))
-  const inputRef = useRef<HTMLInputElement>(null)
+/**
+ * component to read and process QR codes either
+ * by constantly reading image data from a video stream
+ * provided by a camera or by processing image data from
+ * clipboard or an imported image file
+ *
+ * used in Qr dialog (QrCode) in second tab
+ */
+export const QrReader = forwardRef<QrCodeScanRef, Props>(
+  ({ onError, onScanSuccess }: Props, ref) => {
+    const tx = useTranslationFunction()
+    const { openContextMenu } = useContext(ContextMenuContext)
+    const openAlertDialog = useAlertDialog()
+
+    const videoRef = useRef<HTMLVideoElement>(null)
+    const canvasRef = useRef<OffscreenCanvas>(new OffscreenCanvas(640, 480))
+    const inputRef = useRef<HTMLInputElement>(null)
+
+    const [ready, setReady] = useState(false)
+    const [cameraAccessError, setCameraAccessError] = useState(false)
+    const [videoDevices, setVideoDevices] = useState<MediaDeviceInfo[]>([])
+    const [deviceId, setDeviceId] = useState<string | undefined>(undefined)
+    const [processingFile, setProcessingFile] = useState(false)
+
+    const worker = useRef<Worker>(new QrWorker() as Worker)
+    const workerClipBoard = useRef<Worker>(new QrWorker() as Worker)
```

Comment:
This unconditionally instantiates a new worker on every render, resulting in a memory leak.

Maybe see https://react.dev/reference/react/useRef#avoiding-recreating-the-ref-contents on how to resolve.